src/kv/untyped_map_handle.h

```diff
@@ -82,6 +82,48 @@ namespace kv::untyped
       return &search->value;
     }
 
+    template <class F>
+    void foreach_state_and_writes(F&& f, bool always_consider_writes)
```

Comment:
Refactoring this in another private method that lets us early exit once we've passed the end of the range, but still proceed to look at the writes (`always_consider_writes = true`). `always_consider_writes = false` for the existing `foreach` implementation.